src/algebra/algebra/kleene.lean

```diff
@@ -0,0 +1,416 @@
+/-
+Copyright (c) 2022 Siddhartha Prasad. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Siddhartha Prasad.
+-/
+import data.real.basic
+import data.vector
+import tactic.explode
+import tactic.find
+import tactic.induction
+import tactic.rcases
+import tactic.rewrite
+
+
+/-!
+# Kleene Algebras
+
+This file defines Kleene algebras, which are used extensively in theory of computation.
+
+A Kleene algebra is an idempotent semiring with an additional unary operator
+called a Kleene star (⋆).
+
+## References
+
+* (Kozen, D. . A completeness theorem for Kleene algebras and the algebra of regular events.)
+  [https://www.cs.cornell.edu/~kozen/Papers/ka.pdf].
+* https://planetmath.org/idempotentsemiring
+* https://encyclopediaofmath.org/wiki/Idempotent_semi-ring
+* https://planetmath.org/kleenealgebra
+
+
+## Tags
+
+kleene algebra
+
+-/
+
+
+
+namespace i_semiring
+
+universe u
+variables {α : Type u}
+
+/--
+An isemiring is a semiring with the additional property that the addition (+)
+operation is idempotent.
+
+For some idempotent semiring S,  ∀ x ∈ S, `x + x = x`.
+
+Additionally, the binary relation ≤ defines a partial order on isemirings.
+where, a ≤ b ↔ a + b = b
+
+-/
+
+class isemiring  (α : Type u) extends semiring α, semilattice_sup α :=
```

Comment:
The linter was telling you to not put `isemiring` inside the `namespace`; it wasn't telling you to give them different names

Reply:
Sorry, I'm a little confused. Does it want me to put everything inside the `kleene_algebra` namespace?

Reply:
It wants you to write
```lean
class kleene_algebra :=
...

namespace kleene_algebra

lemma some_lemma

end kleene_algebra
```
and not

```lean
namespace kleene_algebra

class kleene_algebra :=
...

lemma some_lemma

end kleene_algebra
```

Reply:
I see! Thank you so much!